client/homebrew/pages/userPage/userPage.jsx

```diff
@@ -1,82 +1,54 @@
 const Account = require('../../navbar/account.navitem.jsx');
 const NewBrew = require('../../navbar/newbrew.navitem.jsx');
 const HelpNavItem = require('../../navbar/help.navitem.jsx');
 const ErrorNavItem = require('../../navbar/error-navitem.jsx');
 const VaultNavitem = require('../../navbar/vault.navitem.jsx');
 
-const UserPage = createClass({
-	displayName     : 'UserPage',
-	getDefaultProps : function() {
-		return {
-			username : '',
-			brews    : [],
-			query    : '',
-			error    : null
-		};
-	},
-	getInitialState : function() {
-		const usernameWithS = this.props.username + (this.props.username.endsWith('s') ? `’` : `’s`);
-
-		const brews = _.groupBy(this.props.brews, (brew)=>{
-			return (brew.published ? 'published' : 'private');
-		});
-
-		const brewCollection = [
-			{
-				title : `${usernameWithS} published brews`,
-				class : 'published',
-				brews : brews.published
-			}
-		];
-		if(this.props.username == global.account?.username){
-			brewCollection.push(
-				{
-					title : `${usernameWithS} unpublished brews`,
-					class : 'unpublished',
-					brews : brews.private
-				}
-			);
-		}
-
-		return {
-			brewCollection : brewCollection
-		};
-	},
-	errorReported : function(error) {
-		this.setState({
-			error
-		});
-	},
-
-	navItems : function() {
-		return <Navbar>
-			<Nav.section>
-				{this.state.error ?
-					<ErrorNavItem error={this.state.error} parent={this}></ErrorNavItem> :
-					null
-				}
-				<NewBrew />
-				<HelpNavItem />
-				<VaultNavitem/>
-				<RecentNavItem />
-				<Account />
-			</Nav.section>
-		</Navbar>;
-	},
-
-	render : function(){
-		return <ListPage brewCollection={this.state.brewCollection} navItems={this.navItems()} query={this.props.query} reportError={this.errorReported}></ListPage>;
-	}
-});
+const UserPage = ({ username = '', brews = [], query = '', error = null, ...props }) => {
+
+    const [currentError, setCurrentError] = useState(error);
+
+    const usernameWithS = username + (username.endsWith('s') ? `’` : `’s`);
+    const groupedBrews = _.groupBy(brews, brew => brew.published ? 'published' : 'private');
+
+	const brewCollection = [
+		{
+			title: `${usernameWithS} published brews`,
+			class: 'published',
+			brews: groupedBrews.published || []
+		},
+		...(username === global.account?.username ? [{
+			title: `${usernameWithS} unpublished brews`,
+			class: 'unpublished',
+			brews: groupedBrews.private || []
+		}] : [])
+	];	
+
+    const navItems = (
+        <Navbar>
+            <Nav.section>
+                {currentError && (<ErrorNavItem error={currentError} parent={null}></ErrorNavItem>)}
+                <NewBrew />
+                <HelpNavItem />
+                <VaultNavitem />
+                <RecentNavItem />
+                <Account />
+            </Nav.section>
+        </Navbar>
+    );
+
+    return (
+        <ListPage brewCollection={brewCollection}  navItems={navItems()} query={query} reportError={(error) =>setCurrentError(error)} />
```

Comment:
With above change, will need to change navItems here too:

```suggestion
        <ListPage brewCollection={brewCollection}  navItems={navItems} query={query} reportError={(error) =>setCurrentError(error)} />
```